core/src/main/java/io/grpc/internal/GrpcUtil.java

```diff
@@ -131,9 +131,9 @@
   public static final String MESSAGE_ACCEPT_ENCODING = "grpc-accept-encoding";
 
   /**
-   * The default maximum uncompressed size (in bytes) for inbound messages. Defaults to 100 MiB.
+   * The default maximum uncompressed size (in bytes) for inbound messages. Defaults to 4 MiB.
```

Comment:
I'm not sure what sort of rationale you are hoping for that isn't time-based. "An arbitrary size that seemed safe based on the users we were aware of"?